modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/global_search/GlobalSearch.js

```diff
@@ -0,0 +1,408 @@
+	const [ordersCount, updateOrdersCount] = useState(null);
+	const [debouncedGetOrders, updateDebouncedGetOrders] = useState(null);
+	const [productsLoading, updateProductsLoading] = useState(false);
+	const [products, updateProducts] = useState(null);
+	const [productsCount, updateProductsCount] = useState(null);
+	const [debouncedGetProducts, updateDebouncedGetProducts] = useState(null);
+	const [active, setActive] = useState(false);
+	const [ids] = useState({
+		input: 'global-search-input' + getRandomId(),
+		menu: 'global-search-menu' + getRandomId(),
+	});
+
+	useEffect(() => {
+		updateDebouncedGetAccounts(() =>
+			debouncePromise(
+				AccountsResource.getAccounts,
+				props.fetchDataDebounce
+			)
+		);
+		updateDebouncedGetOrders(() =>
+			debouncePromise(OrderResource.getOrders, props.fetchDataDebounce)
+		);
+		updateDebouncedGetProducts(() =>
+			debouncePromise(
+				CatalogResource.getProductsByChannelId,
+				props.fetchDataDebounce
+			)
+		);
+	}, [props.fetchDataDebounce]);
+
+	const getProducts = useCallback(() => {
+		composeDataUpdate(
+			debouncedGetProducts(props.channelId, null, {
+				pageSize: props.resultsPageSize,
+				search: query,
+			}),
+			isMounted,
+			updateProducts,
+			updateProductsCount,
+			updateProductsLoading
+		);
+	}, [
+		debouncedGetProducts,
+		props.channelId,
+		props.resultsPageSize,
+		query,
+		isMounted,
+	]);
+
+	const getAccounts = useCallback(() => {
+		composeDataUpdate(
+			debouncedGetAccounts(null, {
+				pageSize: props.resultsPageSize,
+				search: query,
+			}),
+			isMounted,
+			updateAccounts,
+			updateAccountsCount,
+			updateAccountsLoading
+		);
+	}, [debouncedGetAccounts, props.resultsPageSize, query, isMounted]);
+
+	const getOrders = useCallback(() => {
+		composeDataUpdate(
+			debouncedGetOrders(null, {
+				pageSize: props.resultsPageSize,
+				search: query,
+			}),
+			isMounted,
+			updateOrders,
+			updateOrdersCount,
+			updateOrdersLoading
+		);
+	}, [debouncedGetOrders, props.resultsPageSize, query, isMounted]);
+
+	function resetContent() {
+		updateAccounts(null);
+		updateOrders(null);
+		updateProducts(null);
+		updateAccountsCount(null);
+		updateOrdersCount(null);
+		updateProductsCount(null);
+	}
+
+	useEffect(() => {
+		setActive(Boolean(query));
+
+		if (!query) {
+			resetContent();
+
+			return;
+		}
+
+		updateProductsLoading(true);
+		updateAccountsLoading(true);
+		updateOrdersLoading(true);
+
+		getProducts();
+		getAccounts();
+		getOrders();
+	}, [query, getProducts, getAccounts, getOrders]);
+
+	useEffect(() => {
+		function handleClick(e) {
+			if (
+				!(e.target.closest(`#${ids.menu}`) || e.target.id === ids.input)
+			) {
+				setActive(false);
+			}
+		}
+
+		if (active) {
+			document.addEventListener('click', handleClick);
+		}
+		else {
+			document.removeEventListener('click', handleClick);
+		}
+
+		return () => {
+			document.removeEventListener('click', handleClick);
+		};
+	}, [active, ids]);
```

Comment:
You might be able to replace this `useEffect` by `useEventListener` from `frontend-js-web`.